### PR TITLE
fix: Fix tests after qgb changes

### PR DESCRIPTION
**Pull request by SweeXordious**
Attempts  to fix the tests after the QGB related changed

### CODE REVIEW

**Comment by evan-forbes** [comment]
The remaining test should be failing, see https://github.com/celestiaorg/cosmos-sdk/pull/63#issuecomment-1041823169

although, I could swear that we patched this when fixing all the CI, but I can't find where :grimacing:

File: x/staking/client/testutil/suite.go
Comment (evan-forbes) [inline x/staking/client/testutil/suite.go:65]: wow, gas is increased quite a bit. Is this all due to increased storage requirements? 

it's nbd whatever the reason is, as its not an insane increase and won't be used often

**Review by evan-forbes — COMMENTED** [review]
these changes make sense! we should probably document whatever our decision on which approach we're taking in terms of either using this approach or the `SetOrchestrator` approach, along with the factors that went into that decision

File: x/staking/client/testutil/suite.go
Comment (SweeXordious) [reply x/staking/client/testutil/suite.go:65]: It was around 252954. But I  bumped to, 306313  to always pass

**Comment by SweeXordious** [comment]
> some documentation of what decision we want to make.
I am planning to keep a separate celestia-qgb branch containing these changes for a period of time until we're 100% this is the right path, then we document it along with the QGB stuff.
What do you think ?

**Comment by evan-forbes** [comment]
sounds good

File: client/flags/flags.go
Comment (SweeXordious) [inline client/flags/flags.go:81]: It's alright to leave these here? 
I checked also for the other flags, they are defined two times, here and under context also.

File: x/staking/client/cli/flags.go
Comment (SweeXordious) [inline x/staking/client/cli/flags.go:33]: now  they are defined  under general flags  because they're also used in `gentx`, `sim` etc

**Review by SweeXordious — COMMENTED** [review]
@evan-forbes if you're alright with PR, we can merge it.
No need to spend more time digging in history for the fix, I commented that test :sunglasses:

**Comment by evan-forbes** [comment]
ahh found it https://github.com/celestiaorg/cosmos-sdk/pull/109/commits/434b308ce4dc968070c47952d9489ab7e0ade85b

we should add this before merging instead of commenting out imho, as it ensures that we don't forget about it. It probably won't matter, as we will likely upgrade soon, but out of good habit

File: client/flags/flags.go
Comment (evan-forbes) [reply client/flags/flags.go:81]: > they are defined two times, here and under context also

I would assume so. where are the second definitions?

**Review by evan-forbes — APPROVED** [review]
pulling that commit to fix the failing test is kind of optional, but it wouldn't hurt. 

approving with or without

File: client/flags/flags.go
Comment (SweeXordious) [reply client/flags/flags.go:81]: https://github.com/SweeXordious/cosmos-sdk/blob/f53f9d610cc71e0b67dcf423fade44c7ae9e4a94/client/context.go#L57

**Review by adlerjohn — COMMENTED** [review]
utACK

File: client/flags/flags.go
Comment (SweeXordious) [reply client/flags/flags.go:81]: @evan-forbes Let me know if this is alright so that I merge.
Thanks

Comment (evan-forbes) [reply client/flags/flags.go:81]: ahh I see, we actually probably want to remove those, as they are not global flags. Only the global flags for txs get loaded into that context. When we need access to those flags, we should access via parsing the flags in the definition of the cli command

**Pull request merged**
